homework-submission/react-homework/src/Components/DynamicList.js

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from 'react';
+
+class Header extends Component {
+  render() {
+    return (
+      <header>
+        <h3>Todo List</h3>
+        <span className="stats">Type:Dynamic</span>
+      </header>
+    );
+  }
+}
+
+class TodoItem extends Component {
+  render() {
+    return (
+      <div className="todoItem">
+        <span
+          onClick={this.props.click}
+          className={`todoItem-name ${this.props.done ? ' line-through' : ''}`}
+        >
+          {this.props.description}, {this.props.deadline} {this.props.done}
+        </span>
+      </div>
+    );
+  }
+}
+
+class Form extends Component {
+  render() {
+    return (
+      <div>
+        <form onSubmit={this.props.click}>
+          <input type="number" name="id" placeholder="ïd" className="item" />
```

Comment:
Having to enter an id yourself when adding a todo is a bit strange. You can generate an id using code, give it a try.